evennia/contrib/clothing.py

```diff
@@ -0,0 +1,678 @@
+        remove_message = "%s removes %s." % (wearer, self.name)
+        uncovered_list = []
+        
+        # Check to see if any other clothes are covered by this object.
+        for thing in wearer.contents:
+            # If anything is covered by 
+            if thing.db.covered_by == self:
+                thing.db.covered_by = False
+                uncovered_list.append(thing.name)
+        if len(uncovered_list) > 0:
+            remove_message = "%s removes %s, revealing %s." % (wearer, self.name, list_to_string(uncovered_list))
+        # Echo a message to the room
+        if not quiet:
+            wearer.location.msg_contents(remove_message)
+            
+    def at_get(self, getter):
+        """
+        Makes absolutely sure clothes aren't already set as 'worn'
+        when they're picked up, in case they've somehow had their
+        location changed without getting removed.
+        """
+        self.db.worn = False
+    
+class ClothedCharacter(DefaultCharacter):
+    """
+    Character that displays worn clothing when looked at. You can also
+    just copy the return_appearance hook defined below to your own game's
+    character typeclass.
+    """
+    def return_appearance(self, looker):
+        """
+        This formats a description. It is the hook a 'look' command
+        should call.
+
+        Args:
+            looker (Object): Object doing the looking.
+            
+        Notes:
+            The name of every clothing item carried and worn by the character
+            is appended to their description. If the clothing's db.worn value
+            is set to True, only the name is appended, but if the value is a
+            string, the string is appended to the end of the name, to allow
+            characters to specify how clothing is worn.
+        """
+        if not looker:
+            return ""
+        # get description, build string
+        string = "|c%s|n\n" % self.get_display_name(looker)
+        desc = self.db.desc
+        worn_string_list = []
+        clothes_list = get_worn_clothes(self, exclude_covered=True)
+        # Append worn, uncovered clothing to the description
+        for garment in clothes_list:
+            # If 'worn' is True, just append the name
+            if garment.db.worn == True:
+                worn_string_list.append(garment.name)
+            # Otherwise, append the name and the string value of 'worn'
+            elif garment.db.worn:
+                worn_string_list.append("%s %s" % (garment.name, garment.db.worn))
+        if desc:
+            string += "%s" % desc
+        # Append worn clothes.
+        if worn_string_list:
+            string += "|/|/%s is wearing %s." % (self, list_to_string(worn_string_list))
+        else:
+            string += "|/|/%s is not wearing anything." % self
+        return string
+        
+"""
+----------------------------------------------------------------------------
+HELPER FUNCTIONS START HERE
```

Comment:
Common coding standards suggest the helper functions go above the classes that use them - this makes the code easier to read.